internal/server/data/query.go

```diff
@@ -3,42 +3,20 @@ package data
 import (
 	"strings"
 
+	"github.com/infrahq/infra/internal/server/data/querybuilder"
 	"github.com/infrahq/infra/uid"
 )
 
-type queryBuilder struct {
-	query strings.Builder
-	Args  []interface{}
-}
-
-// Query initializes a queryBuilder and returns it populated with stmt.
-// The stmt string can generally contain any SELECT, FROM, or JOIN
-// clauses, and may contain the entire query as long as there are no
-// query parameters.
-// Use queryBuilder.B to add sections of a query with parameters.
-func Query(stmt string) *queryBuilder {
-	q := &queryBuilder{}
-	q.query.WriteString(stmt + " ")
-	return q
-}
-
-// B adds clause and args to the query. The clause must be a trusted string
-// literal. Any arguments must be passed as args so that they are properly
-// escaped by the database driver.
-func (q *queryBuilder) B(clause string, args ...interface{}) {
-	q.query.WriteString(clause + " ")
-	q.Args = append(q.Args, args...)
-}
-
-// String returns the query string, which is used as the first parameter to
-// WriteTxn.Exec, or ReadTxn.Query. You must also pass q.Args as the varargs.
-func (q *queryBuilder) String() string {
-	return q.query.String()
-}
-
 type Table interface {
-	Table() string
-	// Columns returns the names of the tables columns.
+	Table(*querybuilder.Builder)
```

Comment:
By inverting the implementation of the `Table` method like this we can remove it as a special case. The `Table` method can call `Builder.B` like everything else.

I could not find a way to do that with the column functions without having to generate 4 separate methods for every type.